contrib/pkg/sets/sets.go

```diff
@@ -34,18 +34,21 @@ func Key(id ezkube.ResourceId) string {
 	if id == nil {
 		return "<unknown>"
 	}
-	if clusterId, ok := id.(ezkube.ClusterResourceId); ok {
-		b.WriteString(clusterId.GetName())
-		b.WriteString(separator)
-		b.WriteString(clusterId.GetNamespace())
-		b.WriteString(separator)
-		b.WriteString(clusterId.GetClusterName())
-		return b.String()
-	}
 	b.WriteString(id.GetName())
 	b.WriteString(separator)
 	b.WriteString(id.GetNamespace())
 	b.WriteString(separator)
+	if clusterId, ok := id.(ezkube.ClusterResourceId); ok {
+		b.WriteString(separator)
+		// handle the possibility that clusterName could be set either as an annotation (new way)
+		// or as a field (old way pre-k8s 1.25)
+		clusterNameByAnnotation := ezkube.GetClusterName(clusterId)
+		if clusterNameByAnnotation != "" {
+			b.WriteString(clusterNameByAnnotation)
+		} else {
+			b.WriteString(ezkube.GetDeprecatedClusterName(id))
```

Comment:
I don't think this will work as the other object does not conform to `ezkube.ClusterResourceId`